addons/portal/views/portal_templates.xml

```diff
@@ -491,7 +494,7 @@
         </div>
         <div t-attf-class="mb-3 #{error.get('country_id') and 'o_has_error' or ''} col-xl-6">
             <label class="col-form-label" for="country_id">Country</label>
-            <select name="country_id" t-attf-class="form-select #{error.get('country_id') and 'is-invalid' or ''}">
+            <select name="country_id" t-attf-class="form-select #{error.get('country_id') and 'is-invalid' or ''}" t-att-disabled="None if partner_can_edit_vat else '1'">
```

Comment:
It might be useful to also add a message that will be displayed on smaller viewports (`d-xl-none`), like it's already done for `vat` and `company_name`